### PR TITLE
add buildin round_dec

### PR DESCRIPTION
**Pull request by colinback**
What have you changed? (mandatory)

Please explain IN DETAIL what the changes are in this PR and why they are needed:

    add round_dec function

What are the type of the changes? (mandatory)

The currently defined types are listed below, please pick one of the types for this PR by removing the others:

    New feature (non-breaking change which adds functionality)

How has this PR been tested? (mandatory)

    unit test
Does this PR affect documentation (docs/docs-cn) update? (mandatory)

    no
Does this PR affect tidb-ansible update? (mandatory)

    no
Refer to a related PR or issue link (optional)

    https://github.com/tikv/tikv/issues/3275

Benchmark result if necessary (optional)
Add a few positive/negative examples (optional)

### CODE REVIEW

File: src/coprocessor/dag/expr/builtin_math.rs [hunk 1]
```diff
@@ -108,6 +108,16 @@ impl ScalarFunc {
...
     }
 
+    #[inline]
+    pub fn round_dec(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<i64>> {
+        let d = try_opt!(self.children[0].eval_decimal(ctx, row));
+        let d: Result<Decimal> = d
+            .into_owned()
+            .round(DEFAULT_FSP, RoundMode::HalfEven)
+            .into();
+        d.and_then(|dec| dec.as_i64_with_ctx(ctx)).map(Some)
```
Comment (AndreMouche) [inline src/coprocessor/dag/expr/builtin_math.rs]: I think we need to return decimal according to https://dev.mysql.com/doc/refman/5.7/en/mathematical-functions.html#function_round

> The return value has the same type as the first argument (assuming that it is integer, double, or decimal). 

and the related function in tidb https://github.com/pingcap/tidb/blob/master/expression/builtin_math.go#L366

Comment (colinback) [reply src/coprocessor/dag/expr/builtin_math.rs]: @AndreMouche One more question, does it need second parameter D in tikv?
> ROUND(X), ROUND(X,D)
>
> Rounds the argument X to D decimal places. The rounding algorithm depends on the data > type of X. D defaults to 0 if not specified. D can be negative to cause D digits left of the >decimal point of the value X to become zero.

Comment (colinback) [reply src/coprocessor/dag/expr/builtin_math.rs]: @AndreMouche Oh, it seems ROUND(X) are RoundWithFracXXX

**Comment by hicqu** [comment]
LGTM.

**Comment by AndreMouche** [comment]
Please fix the conflicts.

**Comment by siddontang** [comment]
Thanks @colinback 

Please sign off your all commits. See https://github.com/tikv/tikv/blob/master/CONTRIBUTING.md#signing-off-the-commit

If you have already pushed some commits before, you can use `git rebase -i master --signoff` to resign them off.

**Comment by colinback** [comment]
@AndreMouche Any guide document to fix conflict? I delete my branch, rebase master, change code and `git push -f`

**Comment by colinback** [comment]
test_check_args failed?

File: src/coprocessor/dag/expr/scalar_function.rs [hunk 2]
```diff
@@ -187,6 +187,7 @@ impl ScalarFunc {
             | ScalarFuncSig::FloorIntToDec
             | ScalarFuncSig::FloorDecToDec
             | ScalarFuncSig::FloorDecToInt
+            | ScalarFuncSig::RoundDec
```
Comment (AndreMouche) [inline src/coprocessor/dag/expr/scalar_function.rs]: You should update the function `test_check_args` too

**Pull request closed without merging**